### PR TITLE
Resolve #202.

### LINKED ISSUE #202: BlockManager: Block palette.
Required to get runtimeIds working.

### PR DESCRIPTION
**Pull request by filfat**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,7 +64,9 @@
     "@iarna/toml": "^2.2.5",
     "@jsprismarine/bedrock-data": "1.16.20e",
     "@jsprismarine/jsbinaryutils": "2.1.7",
+    "@types/cacache": "^12.0.1",
```
Comment (filfat) [inline package.json:67]: Use —save-dev for types.

File: src/block/BlockManager.ts [hunk 2]
```diff
@@ -82,29 +86,95 @@ export default class BlockManager {
...
 
-            this.getBlocks()
-                .filter((b) => b.meta === 0)
-                .map((block) => () => {
-                    let tag: NBTTagCompound = new NBTTagCompound('');
-                    let nbtBlock: NBTTagCompound = new NBTTagCompound('block');
+        try {
+            this.blockPalette = await this.server
+                .getCacheManager()
+                .getCachedPalette();
```
Comment (filfat) [inline src/block/BlockManager.ts:94]: We need to handle updating the change when the blockdata changes. (Eg new versions or a plugin author registers one).

Comment (HerryYT) [reply src/block/BlockManager.ts:94]: Plugins shouldn't register any block palette. Versioned cache commit is almost finished 👍

Comment (filfat) [reply src/block/BlockManager.ts:94]: @HerryYT They might however register a new block. Which will require a new block palette generation.

Comment (filfat) [reply src/block/BlockManager.ts:94]: E.g. use something like this for the version id `sha256(Array.from(this.blocks).map(a => a.name).sort((a, b) => a - b).join() + prismarine_version)`

Comment (filfat) [reply src/block/BlockManager.ts:94]: That way if `this.blocks` changes we'll automatically generate a new palette.

**Pull request closed without merging**